components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/AbstractResponseTypeHandler.java

```diff
@@ -117,4 +122,48 @@ private void handleErrorRequest(OAuthAuthzReqMessageContext authzReqMessageConte
         authzReqMessageContext.addProperty("ErrorCode", errorCode);
         authzReqMessageContext.addProperty("ErrorMsg", errorMsg);
     }
+
+    @Override
+    public boolean isAuthorizedClient(OAuthAuthzReqMessageContext authzReqMsgCtx) throws IdentityOAuth2Exception {
+
+        OAuth2AuthorizeReqDTO authzReqDTO = authzReqMsgCtx.getAuthorizationReqDTO();
+
+        String grantType = null;
+
+        if (OAuthConstants.CODE.equals(authzReqDTO.getResponseType())) {
+            grantType = GrantTypes.AUTHORIZATION_CODE;
+        } else if (OAuthConstants.TOKEN.equals(authzReqDTO.getResponseType())) {
+            grantType = GrantTypes.IMPLICIT;
+        }
+
+        // Load application data from the cache
+        AppInfoCache appInfoCache = AppInfoCache.getInstance();
+        String consumerKey = authzReqDTO.getConsumerKey();
+        OAuthAppDO oAuthAppDO = appInfoCache.getValueFromCache(consumerKey);
+        if (oAuthAppDO == null) {
+            try {
+                oAuthAppDO = new OAuthAppDAO().getAppInformation(consumerKey);
+                appInfoCache.addToCache(consumerKey, oAuthAppDO);
+            } catch (InvalidOAuthClientException e) {
+                throw new IdentityOAuth2Exception(e.getMessage(), e);
+            }
+        }
```

Comment:
OAuthAppDO is already set as a property in message context. No need to fetch again.